# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/mpegaudiodec_common.c
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/mpegaudiodec_common.c
--- a/media/ffvpx/libavcodec/mpegaudiodec_common.c	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/mpegaudiodec_common.c	Thu Feb 08 10:39:54 2024 +0000
@@ -64,7 +64,7 @@
 };
 
 /* mpegaudio layer 3 huffman tables */
-VLC ff_huff_vlc[16];
+const VLCElem *ff_huff_vlc[16];
 static VLCElem huff_vlc_tables[128 + 128 + 128 + 130 + 128 + 154 + 166 + 142 +
                                204 + 190 + 170 + 542 + 460 + 662 + 414];
 VLC ff_huff_quad_vlc[2];
@@ -401,6 +401,7 @@
 
 static av_cold void mpegaudiodec_common_init_static(void)
 {
+    VLCInitState state = VLC_INIT_STATE(huff_vlc_tables);
     const uint8_t *huff_sym = mpa_huffsymbols, *huff_lens = mpa_hufflens;
     int offset;
 
@@ -414,7 +415,6 @@
     }
 
     /* huffman decode tables */
-    offset = 0;
     for (int i = 0; i < 15;) {
         uint16_t tmp_symbols[256];
         int nb_codes_minus_one = mpa_huff_sizes_minus_one[i];
@@ -426,16 +426,14 @@
             tmp_symbols[j] = high << 1 | ((high && low) << 4) | low;
         }
 
-        ff_huff_vlc[++i].table         = huff_vlc_tables + offset;
-        ff_huff_vlc[i].table_allocated = FF_ARRAY_ELEMS(huff_vlc_tables) - offset;
-        ff_init_vlc_from_lengths(&ff_huff_vlc[i], 7, j,
-                                 huff_lens, 1, tmp_symbols, 2, 2,
-                                 0, INIT_VLC_STATIC_OVERLONG, NULL);
-        offset    += ff_huff_vlc[i].table_size;
+        ff_huff_vlc[++i] = ff_vlc_init_tables_from_lengths(&state, 7, j,
+                                                           huff_lens, 1,
+                                                           tmp_symbols, 2, 2,
+                                                           0, 0);
         huff_lens += j;
         huff_sym  += j;
     }
-    av_assert0(offset == FF_ARRAY_ELEMS(huff_vlc_tables));
+    av_assert1(state.size == 0);
 
     offset = 0;
     for (int i = 0; i < 2; i++) {
@@ -443,9 +441,9 @@
         ff_huff_quad_vlc[i].table = huff_quad_vlc_tables + offset;
         ff_huff_quad_vlc[i].table_allocated = 1 << bits;
         offset                             += 1 << bits;
-        init_vlc(&ff_huff_quad_vlc[i], bits, 16,
+        vlc_init(&ff_huff_quad_vlc[i], bits, 16,
                  mpa_quad_bits[i], 1, 1, mpa_quad_codes[i], 1, 1,
-                 INIT_VLC_USE_NEW_STATIC);
+                 VLC_INIT_USE_STATIC);
     }
     av_assert0(offset == FF_ARRAY_ELEMS(huff_quad_vlc_tables));
 